Refactor upgradeInsecureRequest logic, move to ResourceRequest

sysrqb opened the pull request:
#### 2feb80075a7dd69676d798f0c0363cf48ddd8d1f
<pre>
Refactor upgradeInsecureRequest logic, move to ResourceRequest
<a href="https://bugs.webkit.org/show_bug.cgi?id=276806">https://bugs.webkit.org/show_bug.cgi?id=276806</a>

Reviewed by Brent Fulgham.

upgradeInsecureRequest was implemented in ContentSecurityPolicy because that is
where it is specified by the standard. However, we&apos;re now using (re-using) that
logic in more places. This includes for upgraded mixed-content and when
preferring HTTPS for top-level navigations. For the cases of top-level
navigations, we may not have a Document or CSP policy at the time when we need
to make a decision, so consulting logic in ContentSecurityPolicy is confusing.

This patch moves the relevant logic into ResourceRequestBase which is a more
appropriate place for us to make decisions about modifying a request.

No behavioral changes, no new tests.

* Source/WebCore/contentextensions/ContentExtensionsBackend.cpp:
(WebCore::ContentExtensions::applyResultsToRequest):
* Source/WebCore/page/csp/ContentSecurityPolicy.cpp:
(WebCore::ContentSecurityPolicy::upgradeInsecureRequestIfNeeded const):
* Source/WebCore/platform/network/ResourceRequestBase.cpp:
(WebCore::ResourceRequestBase::upgradeInsecureRequest):
(WebCore::ResourceRequestBase::upgradeInsecureRequestIfNeeded):
(WebCore::ResourceRequestBase::upgradeToHTTPS): Deleted.
* Source/WebCore/platform/network/ResourceRequestBase.h:

Canonical link: <a href="https://commits.webkit.org/281156@main">https://commits.webkit.org/281156@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e0b98b7b8d1be67e83344c07ca108bff74256553

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/58899 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/38228 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/11390 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/62531 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/9343 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/61029 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/45879 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/9544 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/47643 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/6655 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/60930 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/35769 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/50946 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/28500 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/32500 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/8233 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/8347 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/54457 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/8513 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/64232 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/2812 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/8482 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/54958 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/2822 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/50969 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/55056 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/2359 "Passed tests") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/8809 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/34057 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/35141 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/36226 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/34887 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

brentfulgham reviewed (approved):
The change looks reasonable to me. r=me

Commit:
Refactor upgradeInsecureRequest logic, move to ResourceRequest
https://bugs.webkit.org/show_bug.cgi?id=276806

Reviewed by Brent Fulgham.

upgradeInsecureRequest was implemented in ContentSecurityPolicy because that is
where it is specified by the standard. However, we're now using (re-using) that
logic in more places. This includes for upgraded mixed-content and when
preferring HTTPS for top-level navigations. For the cases of top-level
navigations, we may not have a Document or CSP policy at the time when we need
to make a decision, so consulting logic in ContentSecurityPolicy is confusing.

This patch moves the relevant logic into ResourceRequestBase which is a more
appropriate place for us to make decisions about modifying a request.

No behavioral changes, no new tests.

* Source/WebCore/contentextensions/ContentExtensionsBackend.cpp:
(WebCore::ContentExtensions::applyResultsToRequest):
* Source/WebCore/page/csp/ContentSecurityPolicy.cpp:
(WebCore::ContentSecurityPolicy::upgradeInsecureRequestIfNeeded const):
* Source/WebCore/platform/network/ResourceRequestBase.cpp:
(WebCore::ResourceRequestBase::upgradeInsecureRequest):
(WebCore::ResourceRequestBase::upgradeInsecureRequestIfNeeded):
(WebCore::ResourceRequestBase::upgradeToHTTPS): Deleted.
* Source/WebCore/platform/network/ResourceRequestBase.h:

Canonical link: https://commits.webkit.org/281156@main
